bundles/storage/org.eclipse.smarthome.storage.mapdb.test/src/main/java/org/eclipse/smarthome/storage/mapdb/StorageServiceOSGiTest.java

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.storage.mapdb;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.junit.Assert.assertThat;
+
+import java.io.File;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.io.FileUtils;
+import org.eclipse.smarthome.core.storage.Storage;
+import org.eclipse.smarthome.core.storage.StorageService;
+import org.eclipse.smarthome.test.java.JavaOSGiTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author Thomas Eichstaedt-Engelen - Initial contribution
+ * @author Alex Tugarev - Added test for getStorage without classloader
+ * @author Simon Kaufmann - ported to Java
+ */
+public class StorageServiceOSGiTest extends JavaOSGiTest {
+
+    private StorageService storageService;
+    private Storage<PersistedItem> storage;
+
+    @Before
+    public void setUp() {
+        storageService = getService(StorageService.class);
```

Comment:
Shouldn't you use the implementation class attribute, too?
If someone changes the used bundles some time, we still want to ensure that the MapDB one is used.